qcelemental/models/common_models.py

```diff
@@ -1,55 +1,58 @@
 from enum import Enum
-from typing import Any
+from typing import Any, Dict, Optional
 
 import numpy as np
-from pydantic import BaseModel, Extra
+from pydantic import BaseModel
 
 ndarray_encoder = {np.ndarray: lambda v: v.flatten().tolist()}
 
 
 class Provenance(BaseModel):
     creator: str
-    version: str = None
-    routine: str = None
+    version: Optional[str] = None
+    routine: Optional[str] = None
```

Comment:
this means that all three of creator/version/routine must be present in an instance but latter two may be defaulted if not supplied? molparse is demanding all three be supplied.

Reply:
Yes, this was always the case. The `Optional` isn't actually used unless a `mypy --strict-optional` is set. In general, version and routine are always set.

Reply:
ok. so long as we don't hit cases that pass you and fail me, all's well.